### DISCUSSION TITLE
Why are existing fulfillments edited when returns are created?

### DISCUSSION THREAD

**Discussion by rrustvold** [saleor/saleor#7949] [Q&A]
While using the saleor dashboard I noticed that when a return is created from a fulfillment, two things happen:

1. A "return" fulfillment is created. This makes sense.
2. The quantity of the original fulfillment is reduced by whatever the quantity of the return was. 

For example, assume an order includes a quantity of 2 units of a particular item, and that those 2 units were shipped and delivered together as a fulfillment. Then if 1 of those units is returned, it would appear as though the original fulfillment only shipped 1 item, when it actually shipped 2. 

This doesn't seem like a great idea to me, especially since the return record has no connection to the fulfillment it was created from. If someone was to look up a fulfillment or tracking number, they might incorrectly assume that only 1 unit was shipped as part of a delivery. 

I would think that it would be wiser to not edit the existing fulfillment, and only create the return "fulfillment" when a return is processed. Furthermore if an entire fulfillment is returned, then it removes it entirely, as if it never happened. This loses the tracking number which could be undesirable.

**Comment by pwgryglak** [comment]
Hi! Thanks for asking.
We wanted to have a consistent number of items in the order so our users will have a clear view of what is happening with the items in the order. It might be misleading if we would have 3 items fulfilled and 3 items returned - Would it mean that we already returned those 3 items or did we return 3 and 3 are still shipped and at the customer’s house? Duplicating items might confuse users when the order contained 3 items and we end up showing more.
Creating some connection between the return and fulfillment is a good idea, and we will look into that when our roadmap allows us to.
Thanks for your feedback!

**Reply by patrys** [reply]
I've talked about this previously, and I think we should have "ghost rows" that indicate where an item lived originally (additional semi-transparent rows at the end of the source fulfillment that once clicked will show you where the items live now).

**Reply by pwgryglak** [reply]
Makes sense - I'll provide the designs
